internal/pkg/service/buffer/worker/task/node.go

```diff
@@ -0,0 +1,184 @@
+// Package task provides a task abstraction for long-running operations in the Worker node.
+// It is guaranteed that the task will run at most once, as well as resistance to outages.
+package task
```

Comment:
Aby  bol kod lepsie udrziavatelny, tak som vytvoril tuto `task` abstrakciu, ak potrebujeme aby nieco bezalo iba raz. Takto sa to da oddelene otestovat na jednom mieste.

Je to doplnok ku `distribution.Node`, ktora v `99%` pripadov rozdeli pracu spravne, ale pri zmenach v `cluster` moze existovat kratky okamich, ked nemaju vsetky nodes rovnake informacie.

Vid komentar:
https://github.com/keboola/keboola-as-code/blob/8b46279d5fb66532c82942f2e67a2fd3d2baa0dc/internal/pkg/service/buffer/worker/distribution/node.go#L11-L18

To riesi tento `task` package, kedze pouziva zamok/transkaciu + lease.

-----

- Ak by sme nemali zamok, tak sa moze stat, ze nejaka operacia bezi duplicitne.
- Ak by sme nemali `hashring` v `distribution.Node`, tak by sa to riesilo cele iba cez zamky, a dochadzalo by ku castym koliziam -> bolo by to narocne na resources.
- Takto mame obe, aj garancie, aj nenarocnost na resources.